bindings/cpp/session.cpp

```diff
@@ -2249,7 +2264,46 @@ async_iterator_result session::start_iterator(const key &id, const std::vector<d
 	req->time_end = time_end;
 	req->range_num = ranges.size();
 
-	memcpy(data.skip<dnet_iterator_request>().data(), &ranges.front(), ranges_size);
+	if (ranges_size)
+		memcpy(data.skip<dnet_iterator_request>().data(), &ranges.front(), ranges_size);
+
+	return iterator(id, data);
+}
+
+async_iterator_result session::start_copy_iterator(const key &id,
+		const std::vector<dnet_iterator_range>& ranges,
+		uint64_t flags,
```

Comment:
Actually every flag is allowed here. Flags like range/ts-range will be checked in `dnet_iterator_start()`, data and no-meta flags are also allowed